packages/syft/src/syft/core/node/new/user_code.py

```diff
@@ -252,7 +397,8 @@ class UserCode(SyftObject):
     user_unique_func_name: str
     code_hash: str
     signature: inspect.Signature
-    status: UserCodeStatus = UserCodeStatus.SUBMITTED
+    status: UserCodeStatusContext
```

Comment:
Why has the UserCodeStatus changed to `UserCodeStatusContext`? Is this for multiple permissions?

Reply:
Yep, this for multiple permissions

Each UserCodeStatusContext contains a dict, for the parties it requires approval from....

For one domain scenario:

```py
status = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : UserCodeStatus.SUBMITTED,
 }
```

For enclave scenario:

```py
status = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : UserCodeStatus.SUBMITTED,
                  UserNodeView(name="Italy", verify_key= italy_domain.verify_key) : UserCodeStatus.SUBMITTED,
 }
```